### DISCUSSION TITLE
What if max rows of fetched data is over 1000000?

### DISCUSSION THREAD

**Discussion by hayatshin** [supabase/supabase#26215] [Questions]
I realize that there's max rows set of API to fetch data from supabase.
It was set as 1000, so I need to remove it since I have to calculate with fetched data's length.
but max of max rows setting is 1000000.

then what if data length is over 1000000?
Is there any way to remove max rows setting?

**Comment by GaryAustin1** [comment]
Supabase is limiting that to 1000000.  You would need to discuss with support if it can be changed for your instance.

But, it seems like a bad idea to download a million rows for performance and egress cost reasons.  Can you do this calculation in SQL in a postgres function and return the result instead?

**Reply by hayatshin** [reply]
If I deploy SQL postgres function and user `rpc`, then there's no max rows limit and even cost is not charged?

**Reply by GaryAustin1** [reply]
If you can do your operation in the function then the only egress is only the result you return from the fucntion.  You will use a bit of extra load on the CPU/memory doing the work versus just the query but that is not a direct charge, just can the database handle the computation overhead.  There are also SQL aggregate functions and the like designed to deal with entire tables.
